src/rules/spaceWithinParensRule.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { forEachToken } from "tsutils";
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+interface Options {
+    size: number;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "space-within-parens",
+        description: "Enforces spaces within parentheses or disallow them.",
+        hasFix: true,
+        optionsDescription: Lint.Utils.dedent`
+            You may enforce the amount of whitespace within parentheses.
+        `,
+        options: { type: "number", min: 0 },
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_NO_SPACE = "No whitespace within parentheses";
+    public static FAILURE_NEEDS_SPACE = "Needs whitespace within parentheses";
+    public static FAILURE_NO_EXTRA_SPACE(count: number): string {
+        return `No more than ${count} whitespace within parentheses allowed`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new SpaceWhitinParensWalker(sourceFile, this.ruleName, parseOptions(Number(this.ruleArguments[0]))));
+    }
+}
+
+function parseOptions(whitespaceSize?: number): Options {
+    let size = 0;
+    if (typeof whitespaceSize === "number") {
+        if (whitespaceSize >= 0) {
+            size = whitespaceSize;
+        }
+    } else if (whitespaceSize !== undefined) {
+        const parsedSize = parseInt(whitespaceSize, 10);
+        if (!Number.isNaN(parsedSize) && parsedSize >= 0) {
+            size = parsedSize;
+        }
+    }
+    return {
+        size,
+    };
+}
+
+class SpaceWhitinParensWalker extends Lint.AbstractWalker<Options> {
+    public walk(sourceFile: ts.SourceFile) {
+        return ts.forEachChild(sourceFile, (node: ts.Node): void => {
+            this.checkNode(node);
+        });
+    }
+
+    private checkNode(node: ts.Node): void {
+        forEachToken(node, (token: ts.Node) => {
```

Comment:
I actually meant to replace the `ts.forEachChild(...)` in `walk` with this call.
You can just take this method's body and put it into `walk`.

With the current implementation the rule will visit nested tokens multiple times.